Tests/test_counter_BBid_coverage.py

```diff
@@ -0,0 +1,78 @@
+import inspect
+import os
+
+
+def get_full_path(filename):
+    current_file_path = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+    return os.path.join(current_file_path, filename)
+
+
+def calculate_coverage_for_file(cpp_file, test_input):
+    llvm_full_path = get_full_path('../afl-llvm-pass.so')
+    afl_rt_path = get_full_path("../afl-llvm-rt.o")
+    bb_coverage_path = get_full_path('../counter_BBid_coverage')
+
+    bc_file = "%s.o.bc" % cpp_file
+    object_file = "%s.o" % cpp_file
+    coverage_file = './test_bb_coverage_out'
+
+    os.system("wllvm %s -O3 -o %s.o -lstdc++" %(cpp_file, cpp_file))
+    os.system("extract-bc %s.o" % cpp_file)
+    os.system('clang  -Xclang -load -Xclang %s %s %s -o %s > /dev/null' % (llvm_full_path, bc_file, afl_rt_path, object_file))
+    os.system('%s -o %s  -A %s -- %s @@ >/dev/null' %(bb_coverage_path, coverage_file, test_input, object_file))
+
+    os.remove(object_file)
+    os.remove(bc_file)
+
+    block_coverage = {}
+    with open(coverage_file, 'r') as coverage_logs:
+        for l in coverage_logs:
+            coverage_info = l.strip().split(':')
+            assert len(coverage_info)==2, 'Coverage format file should be block:coverage count'
+            block, coverage = int(coverage_info[0]), int(coverage_info[1])
+            block_coverage[block] = coverage
+
+    os.remove(get_full_path('test_bb_coverage_out'))
+    return block_coverage
+
+
+def check_block_id_range(block_coverage, total_blocks):
+    for i in block_coverage:
+        assert 0 <= i <= total_blocks, "Block ID should be in the range of 0 to %d" % total_blocks
+
+
+def test_coverage():
+    test_program = get_full_path('test_program/test_instr.c')
+
+    test_input = get_full_path('test_inputs/example1.txt')
+    block_coverage = calculate_coverage_for_file(test_program, test_input)
+    check_block_id_range(block_coverage, 10)
+    assert len(block_coverage) < 11, 'The test_instr.c file should only have 10 blocks.'
```

Comment:
shall we change the assert `to == 10`?

Reply:
Since the code only writes info for the blocks with non-zero coverage, it might not always be 10.

Reply:
The assumption that it's always less than 11 holds for all the test-cases, which is why I wanted to give a specific error if it's more than that. I can add another assertion for test-specific block_coverage counts (e.g. one test might have 8 covered blocks, the other one might have 4)

Reply:
yes that'd be great!